pyiceberg/table/__init__.py

```diff
@@ -974,7 +974,7 @@ def name_mapping(self) -> NameMapping:
         else:
             return create_mapping_from_schema(self.schema())
 
-    def append(self, df: pa.Table) -> None:
+    def append(self, df: pa.Table, **snapshot_properties) -> None:
```

Comment:
I can see that we want to add more keyword arguments later on, how about:
```suggestion
    def append(self, df: pa.Table, snapshot_properties: Dict[str, str] = EMPTY_DICT) -> None:
```

Same goes for the rest of the API's

Reply:
Sounds good, I was worried about evolving API as well. Thanks!